### PR TITLE
Update shippings settings phase 1

### PR DESCRIPTION
**Pull request by ianlin**
### Changes proposed in this Pull Request:

Project: pcTzPl-2qP-p2

This is the umbrella PR that contains all the changes for shipping phase 1 implementation. The PRs includes:

- #2520
- #2526
- #2527
- #2590
- #2594
- #2609
- #2619
- #2686

### Detailed test instructions:

All the PRs have already been reviewed and tested, but it might be good to have an extra set of eyes to double check everything works as expected.

### Changelog entry

> Update - New shipping settings: phase 1

### CODE REVIEW

**Comment by puntope** [comment]
Do we need to update the [applicable version here in the migration file](https://github.com/woocommerce/google-listings-and-ads/pull/2699/files#diff-898d6eb44e5ad0a4ef4f1b8ac7af29f6d561f5ddfb30b1ab79398dd81a629df8R45-R46)?

**Review by puntope — DISMISSED** [review]

**Review by puntope — COMMENTED** [review]
E2E tests seem to fail 

https://github.com/woocommerce/google-listings-and-ads/actions/runs/12087559628/job/33709246270

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/add-time-button/add-time-modal/index.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.gla-add-time-modal .gla-countries-time-stepper .components-input-control {
+	margin-bottom: 0;
+}
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/add-time-button/add-time-modal/index.scss]: This style may not be necessary as it doesn't seem to be set by other styles.

image

If the style is removed, the following lines can be removed together:

https://github.com/woocommerce/google-listings-and-ads/blob/c1dffd305b34192e0d6daeb0e36571565c86827d/js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/add-time-button/add-time-modal/index.js#L18

https://github.com/woocommerce/google-listings-and-ads/blob/c1dffd305b34192e0d6daeb0e36571565c86827d/js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/add-time-button/add-time-modal/index.js#L77

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/countries-time-input/edit-time-button/index.scss [hunk 2]
```diff
@@ -5,3 +5,7 @@
 		padding: 0;
 	}
 }
+
+.gla-edit-time-modal .gla-countries-time-stepper .components-input-control {
+	margin-bottom: 0;
+}
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/countries-time-input/edit-time-button/index.scss]: This style may not be necessary as it doesn't seem to be set by other styles.

image

The following line can be removed together:

https://github.com/woocommerce/google-listings-and-ads/blob/c1dffd305b34192e0d6daeb0e36571565c86827d/js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/countries-time-input/edit-time-button/edit-time-modal/index.js#L80

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js [hunk 3]
```diff
@@ -7,15 +12,18 @@
  * const shippingTimes = [
  *     {
  *         countryCode: 'US',
- *         time: 10
+ *         time: 10,
+ *		   maxTime: 20,
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js]: Avoid mixed indentation chars.

```suggestion
 *         maxTime: 20,
```

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js [hunk 4]
```diff
@@ -7,15 +12,18 @@
...
  *     },
  *     {
  *         countryCode: 'AU',
- *         time: 10
+ *         time: 10,
+ *		   maxTime: 20,
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js]: Avoid mixed indentation chars.

```suggestion
 *         maxTime: 20,
```

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js [hunk 5]
```diff
@@ -7,15 +12,18 @@
...
  *     },
  *     {
  *         countryCode: 'CN',
- *         time: 15
+ *         time: 15,
+ *		   maxTime: 22,
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js]: Avoid mixed indentation chars.

```suggestion
 *         maxTime: 22,
```

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js [hunk 6]
```diff
@@ -26,10 +34,12 @@
  * //     {
  * //         countries: ['US', 'AU'],
  * //         time: 10
+ *  //		  maxTime: 20,
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js]: Avoid mixed indentation chars.

```suggestion
 * //         maxTime: 20,
```

File: js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js [hunk 7]
```diff
@@ -26,10 +34,12 @@
...
  * //     },
  * //     {
  * //         countries: ['CN'],
  * //         time: 15
+ * //		  maxTime: 22,
```
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/getCountriesTimeArray.js]: Avoid mixed indentation chars.

```suggestion
 * //         maxTime: 22,
```

File: js/src/data/actions.js [hunk 8]
```diff
@@ -228,6 +228,7 @@ export function* fetchShippingTimes() {
 			return {
 				countryCode: el.country_code,
 				time: Number( el.time ),
+				maxTime: Number( el.max_time ),
```
Comment (eason9487) [inline js/src/data/actions.js:232]: The new property is missing from the `ShippingTime` type.

https://github.com/woocommerce/google-listings-and-ads/blob/c1dffd305b34192e0d6daeb0e36571565c86827d/js/src/data/actions.js#L209-L215

File: js/src/data/actions.js [hunk 9]
```diff
@@ -261,14 +262,15 @@ export function* fetchShippingTimes() {
  * @throws Will throw an error if the request failed.
  */
 export function* upsertShippingTimes( shippingTime ) {
-	const { countryCodes, time } = shippingTime;
+	const { countryCodes, time, maxTime } = shippingTime;
```
Comment (eason9487) [inline js/src/data/actions.js:267]: The new property is missing from the `AggregatedShippingTime` type.

https://github.com/woocommerce/google-listings-and-ads/blob/c1dffd305b34192e0d6daeb0e36571565c86827d/js/src/data/actions.js#L250-L256

**Review by eason9487 — COMMENTED** [review]
Had a quick look at the frontend code and wanted to report a few minor issues that could be fixed quickly.

**Comment by ianlin** [comment]
@puntope 

> Do we need to update the [applicable version here in the migration file](https://github.com/woocommerce/google-listings-and-ads/pull/2699/files#diff-898d6eb44e5ad0a4ef4f1b8ac7af29f6d561f5ddfb30b1ab79398dd81a629df8R45-R46)?

I think normally it would be updated manually when doing the release. (See the 5th checkbox in [the release PR](https://github.com/woocommerce/google-listings-and-ads/pull/2694)).

Also wanted to discuss maybe we may want to release this new shipping phase 1 as `2.10.0`?

> E2E tests seem to fail

Thanks for testing it, I missed checking e2e part. I fixed the e2e related to shipping phase 1, however, the [e2e test still failed using the branch of this PR](https://github.com/woocommerce/google-listings-and-ads/actions/runs/12112506569/job/33766544102). It also [failed when I ran e2e on the develop branch](https://github.com/woocommerce/google-listings-and-ads/actions/runs/12114771883/job/33771877427). Given the G4W release time is tomorrow, I will create a follow up issue regarding the e2e tests and we can check that after it release. WDYT?

**Comment by ianlin** [comment]
@eason9487 

> Had a quick look at the frontend code and wanted to report a few minor issues that could be fixed quickly.

Thanks for catching them, I've updated the code regarding your suggestions.

**Comment by puntope** [comment]
@ianlin I approved https://github.com/woocommerce/google-listings-and-ads/pull/2705#pullrequestreview-2472178535 That seems to be fixing the E2E

However- there are some conflicts

**Review by puntope — DISMISSED** [review]

**Review by eason9487 — APPROVED** [review]
Thanks for the additional adjustments.

> It also [failed when I ran e2e on the develop branch](https://github.com/woocommerce/google-listings-and-ads/actions/runs/12114771883/job/33771877427).

E2E on the `develop` branch has been fixed by #2705. I also merged it into this PR and resolved the code conflict in https://github.com/woocommerce/google-listings-and-ads/pull/2699/commits/48fb03c8181edb26cc9627c1593c90eb10173a11.

**Comment by ianlin** [comment]
Thanks @eason9487 and @puntope's team work, merging it now.

**Pull request merged**